### ISSUE TITLE
Would FlexAttention be useful over SDPA for masked language modeling?

### ISSUE THREAD

**Issue by abdulfatir** [pytorch-labs/attention-gym#129]
Thank you for the examples in this repo. I am wondering if FlexAttention would bring any value over SDPA in the case of masked language modeling? In masked language modeling, random input positions are masked and their contribution is ignored during softmax. However, since the masking is random, there's no structure to potentially exploit. Does FlexAttention make sense in this case?

**Comment by drisspg** [comment]
It seems less likely in terms of perf, but you may be able to get some memory savings. If you can come up w/ a compact way to represent your masking as opposed to a full attention_bias
